addon/mixins/touch-action.js

```diff
@@ -2,59 +2,43 @@ import Ember from 'ember';
 
 const {
   computed,
-  K,
-  get,
   Mixin,
-  set,
   String: { htmlSafe }
 } = Ember;
 
-const SafeTouchAction = htmlSafe('touch-action: manipulation; -ms-touch-action: manipulation; cursor: pointer;');
-const SafeEmptyString = htmlSafe('');
-const FocusableInputTypes = ['button', 'submit', 'text', 'file'];
-const TouchActionSelectors = ['button', 'input', 'a', 'textarea'];
-
 export default Mixin.create({
-  touchActionSelectors: TouchActionSelectors,
-  touchActionProperties: SafeTouchAction,
 
   init() {
     this._super(...arguments);
-
-    if (this.touchActionProperties !== SafeTouchAction) {
-      this.touchActionProperties = htmlSafe(this.touchActionProperties);
-    }
-
-    if (this.tagName !== '' && this.click !== K) {
-      let newAttributeBindings = [];
-      const bindings = get(this, 'attributeBindings');
-
-      if (Array.isArray(bindings)) {
-        newAttributeBindings = newAttributeBindings.concat(bindings);
-      }
-
-      newAttributeBindings.push('touchActionStyle:style');
-      set(this, 'attributeBindings', newAttributeBindings);
+    const tagName = this.get('tagName');
+    const hasClick = !!this.get('click');
+    const hasTag = (typeof tagName === 'string' && tagName.length > 0) || (tagName === null && hasClick);
+    if (!hasTag) { return; }
+
+    this.reopen({
+      attributeBindings: ['touchActionStyle:style']
+    });
```

Comment:
fwiw - this is the hack that makes this work - i saw why to some extent - but we should notify core team - i dont think regression is near the call site for merged/concatenated/apply mixin code paths in ember source - it seems more disconnected than that (either instantiation of components(lazy or not) or (hopefully not) basic object related (im newb though, just noting what i saw debugging earlier) for example - tagName and element do not exist for elements during some phases of lifecycle, causing often faux 'cant add attribute bindings to a tagless component' error - meaning in most cases, that error is wrong - eventually all components(that dont set tagName to `''` get a tagName, element and id), just not at time of extension via normal reopen of prototype like used to happen - thus my assumption is this hack to works b/c we reopen prototype during/after Class construction, at which point the prototype for the new instance has been assigned to the constructor we're actively invoking - anyway - those are my observation, naive as they may be - but thought i'd share cause i won't remember later :) )